standard/table.lua

```diff
@@ -63,6 +63,23 @@ function Table.filter(tbl, predicate, argument)
 	return filteredTbl
 end
 
+---@generic K, V, T
+---@param tbl {[K]: V}
+---@param predicate fun(key?: K, value?: V, argument?: T): boolean
+---@param argument T?
+---@return {[K]: V}
+function Table.filterByKey(tbl, predicate, argument)
```

Comment:
What's the use case of argument, i.e. why not integrate it into the predicate?

Reply:
copy paste from the `Table.filter` function
open for kicking it